### PR TITLE
New Resource: Waypoint TFC Config

### PR DESCRIPTION
**Pull request by catsby**
### :building_construction: Acceptance tests

- [x] Are there any feature flags that are required to use this functionality? (No)
- [x] Have you added an acceptance test for the functionality being added? (Yes)
- [x] Have you run the acceptance tests on this branch? (Yes, locally)

Output from acceptance testing:

```sh
$ make testacc TESTARGS="-run=TestAccWaypointTfcConfig_basic"
=== RUN   TestAccWaypointTfcConfig_basic
--- PASS: TestAccWaypointTfcConfig_basic (2.80s)
```

### CODE REVIEW

File: internal/provider/waypoint/resource_waypoint_tfc_config_test.go
Comment (catsby) [inline internal/provider/waypoint/resource_waypoint_tfc_config_test.go:121]: Note that at this time Waypoint does not verify the authenticity of the token or the TFC Org Name *at the time we create the resource*. Those values are used in later features and will fail if not valid, but not when we create the resource.

**Review by manish-hashicorp — APPROVED** [review]
Rubber stamping. Please get approval from another team member before merging.

**Comment by catsby** [comment]
> Rubber stamping. Please get approval from another team member before merging.

Thanks @manish-hashicorp . Are you requesting another approval from someone on the team that owns this provider, or from another member of the Waypoint team? Thanks :D

**Comment by manish-hashicorp** [comment]
Sorry should've been clear. I meant another approval from **Waypoint** team.

File: .github/CODEOWNERS
Comment (HenryEstberg) [inline .github/CODEOWNERS:35]: Non blocking comment: Github claims that the core-sre team is not a valid team to own these two files and so it says that there are errors on line 34 and 35 but you didn't touch those lines so I assume the issue is pre-existing or just github making a mistake.

**Review by HenryEstberg — APPROVED** [review]
LGTM

File: .github/CODEOWNERS
Comment (catsby) [reply .github/CODEOWNERS:35]: I assume this is pre-existing as well. I'm not sure why.

Comment (manish-hashicorp) [reply .github/CODEOWNERS:35]: Core-SRE team has dissolved. We need to find new owner. Kindly ignore for now.

**Pull request merged**
